<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/assets/logos/SN_web_lightmode.png" width="300" alt="SN_web_lightmode.png logo"/>
</center>

# Investigation of diabetes patients readmission among US hospitals

# Lab 1 Dataset Creation

Estimated time needed: **10** minutes

## Objectives

After completing this lab you will be able to:

*   Acquire data in various ways
*   Obtain insights from data with Pandas library


<h2>Table of Contents</h2>

<div class="alert alert-block alert-info" style="margin-top: 20px">
<ol>
    <li><a href="https://#data_acquisition">Data Acquisition</a>
    <li><a href="https://#data_acquisition">Data Acquisition</a>
    <li><a href="https://#basic_insight">Basic Insight of Dataset</a></li>
</ol>

</div>
<hr>


<h1 id="data_acquisition">Data Acquisition</h1>
<p>
There are various formats for a dataset: .csv, .json, .xlsx  etc. The dataset can be stored in different places, on your local machine or sometimes online.<br>

In this section, you will learn how to load a dataset into our Jupyter Notebook.<br>

In our case, the Diabetes Dataset is an online source, and it is in a CSV (comma separated value) format. Let's use this dataset as an example to practice data reading.

<ul>
    <li>Data source: <a href="https://www.kaggle.com/datasets/brandao/diabetes" target="_blank">https://www.kaggle.com/datasets/brandao/diabetes</a></li>
    <li>Data type: csv</li>
</ul>

The statistical data obtained from https://www.kaggle.com/datasets/brandao/diabetes under [CC0: Public Domain](https://creativecommons.org/publicdomain/zero/1.0/) license.

The Pandas Library is a useful tool that enables us to read various datasets into a dataframe; our Jupyter notebook platforms have a built-in <b>Pandas Library</b> so that all we need to do is import Pandas without installing.
</p>


you are running the lab in your  browser, so we will install the libraries using `piplite`


In [3]:
#you are running the lab in your  browser, so we will install the libraries using ``piplite``
#import piplite
#import micropip
#await piplite.install(['pandas'])
#await piplite.install(['matplotlib'])
#await piplite.install(['scipy'])
#await piplite.install(['seaborn'])
#await micropip.install(['ipywidgets'],keep_going=True)
#await micropip.install(['tqdm'],keep_going=True)


If you run the lab locally using Anaconda, you can load the correct library and versions by uncommenting the following:


In [4]:
#install specific version of libraries used in  lab
#! mamba install pandas==1.3.3  -y
#! mamba install numpy=1.21.2 -y

In [5]:
# import pandas library
import pandas as pd
import numpy as np

This function will download the dataset into your browser


In [6]:
#This function will download the dataset into your browser 

#from pyodide.http import pyfetch
#
#async def download(url, filename):
#    response = await pyfetch(url)
#    if response.status == 200:
#        with open(filename, "wb") as f:
#            f.write(await response.bytes())

<h2>Read Data</h2>
<p>
We use <code>pandas.read_csv()</code> function to read the csv file. In the brackets, we put the file path along with a quotation mark so that pandas will read the file into a dataframe from that address. The file path can be either an URL or your local file address.<br>

You can also assign the dataset to any variable you create.

</p>


In [7]:
path = "diabetic_data.csv"

you will need to download the dataset; if you are running locally, please comment out the following


In [8]:
#you will need to download the dataset; if you are running locally, please comment out the following 
# await download(path, "diabetic_data.csv")
# path="diabetic_data.csv"

In [9]:
# Import pandas library
import pandas as pd

# Read the online file by the URL provides above, and assign it to variable "df"

df = pd.read_csv(path)

After reading the dataset, we can use the <code>dataframe.head(n)</code> method to check the top n rows of the dataframe, where n is an integer. Contrary to <code>dataframe.head(n)</code>, <code>dataframe.tail(n)</code> will show you the bottom n rows of the dataframe.


In [10]:
# show the first 5 rows using dataframe.head() method
print("The first 5 rows of the dataframe") 
df.head(5)

The first 5 rows of the dataframe


encounter_id  patient_nbr             race  gender      age weight  \
0       2278392      8222157        Caucasian  Female   [0-10)      ?   
1        149190     55629189        Caucasian  Female  [10-20)      ?   
2         64410     86047875  AfricanAmerican  Female  [20-30)      ?   
3        500364     82442376        Caucasian    Male  [30-40)      ?   
4         16680     42519267        Caucasian    Male  [40-50)      ?   

   admission_type_id  discharge_disposition_id  admission_source_id  \
0                  6                        25                    1   
1                  1                         1                    7   
2                  1                         1                    7   
3                  1                         1                    7   
4                  1                         1                    7   

   time_in_hospital  ... citoglipton insulin  glyburide-metformin  \
0                 1  ...          No      No                   No   
1                 3  ...          No      Up                   No   
2                 2  ...          No      No                   No   
3                 2  ...          No      Up                   No   
4                 1  ...          No  Steady                   No   

   glipizide-metformin  glimepiride-pioglitazone  metformin-rosiglitazone  \
0                   No                        No                       No   
1                   No                        No                       No   
2                   No                        No                       No   
3                   No                        No                       No   
4                   No                        No                       No   

   metformin-pioglitazone  change diabetesMed readmitted  
0                      No      No          No         NO  
1                      No      Ch         Yes        >30  
2                      No      No         Yes         NO  
3                      No      Ch         Yes         NO  
4                      No      Ch         Yes         NO  

[5 rows x 50 columns]

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
    <h1> Question #1: </h1>
    <b>Check the bottom 10 rows of data frame "df".</b>
</div>


In [11]:
# Write your code below and press Shift+Enter to execute

<details><summary>Click here for the solution</summary>

```python
    print("The last 10 rows of the dataframe\n")
    df.tail(10)
```

<h3>Add Headers</h3>
<p>
Take a look at our dataset. Pandas automatically set the header with an integer starting from 0.
</p>
<p>
To better describe our data, we can introduce a header. This information is available at:  <a href="https://www.kaggle.com/datasets/brandao/diabetes" target="_blank">https://www.kaggle.com/datasets/brandao/diabetes</a>.
</p>
<p>
To make our header look better, we can override it.
</p>
<p>
First, we create a list "headers" that include all column names in order.
Then, we use <code>dataframe.columns = headers</code> to replace the headers with the list we created.
</p>

In [12]:
# create headers list
headers = ["Encounter Id","Patient Number","Race","Gender","Age","Weight","Admission Type Id",
           "Discharge Disposition Id","Admission Source Id","Time In Hospital","Payer Code",
           "Medical Specialty","Num Lab Procedures","Num Procedures","Num Medications",
           "Number Outpatient","Number Emergency","Number Inpatient","Diagnosis1","Diagnosis2",
           "Diagnosis3","Number Diagnoses","Max Glu Serum","A1c Result","Metformin","Repaglinide",
           "Nateglinide","Chlorpropamide","Glimepiride","Acetohexamide","Glipizide",
           "Glyburide","Tolbutamide","Pioglitazone","Rosiglitazone","Acarbose","Miglitol",
           "Troglitazone","Tolazamide","Examide","Citoglipton","Insulin","Glyburide-metformin",
           "Glipizide-metformin","Glimepiride-pioglitazone","Metformin-rosiglitazone","Metformin-pioglitazone",
           "Change","Diabetes Medication","Readmitted"]
print("headers\n", headers)

headers
 ['Encounter Id', 'Patient Number', 'Race', 'Gender', 'Age', 'Weight', 'Admission Type Id', 'Discharge Disposition Id', 'Admission Source Id', 'Time In Hospital', 'Payer Code', 'Medical Specialty', 'Num Lab Procedures', 'Num Procedures', 'Num Medications', 'Number Outpatient', 'Number Emergency', 'Number Inpatient', 'Diagnosis1', 'Diagnosis2', 'Diagnosis3', 'Number Diagnoses', 'Max Glu Serum', 'A1c Result', 'Metformin', 'Repaglinide', 'Nateglinide', 'Chlorpropamide', 'Glimepiride', 'Acetohexamide', 'Glipizide', 'Glyburide', 'Tolbutamide', 'Pioglitazone', 'Rosiglitazone', 'Acarbose', 'Miglitol', 'Troglitazone', 'Tolazamide', 'Examide', 'Citoglipton', 'Insulin', 'Glyburide-metformin', 'Glipizide-metformin', 'Glimepiride-pioglitazone', 'Metformin-rosiglitazone', 'Metformin-pioglitazone', 'Change', 'Diabetes Medication', 'Readmitted']


In [13]:
df.columns = headers
df.head(10)

Encounter Id  Patient Number             Race  Gender       Age Weight  \
0       2278392         8222157        Caucasian  Female    [0-10)      ?   
1        149190        55629189        Caucasian  Female   [10-20)      ?   
2         64410        86047875  AfricanAmerican  Female   [20-30)      ?   
3        500364        82442376        Caucasian    Male   [30-40)      ?   
4         16680        42519267        Caucasian    Male   [40-50)      ?   
5         35754        82637451        Caucasian    Male   [50-60)      ?   
6         55842        84259809        Caucasian    Male   [60-70)      ?   
7         63768       114882984        Caucasian    Male   [70-80)      ?   
8         12522        48330783        Caucasian  Female   [80-90)      ?   
9         15738        63555939        Caucasian  Female  [90-100)      ?   

   Admission Type Id  Discharge Disposition Id  Admission Source Id  \
0                  6                        25                    1   
1                  1                         1                    7   
2                  1                         1                    7   
3                  1                         1                    7   
4                  1                         1                    7   
5                  2                         1                    2   
6                  3                         1                    2   
7                  1                         1                    7   
8                  2                         1                    4   
9                  3                         3                    4   

   Time In Hospital  ... Citoglipton Insulin  Glyburide-metformin  \
0                 1  ...          No      No                   No   
1                 3  ...          No      Up                   No   
2                 2  ...          No      No                   No   
3                 2  ...          No      Up                   No   
4                 1  ...          No  Steady                   No   
5                 3  ...          No  Steady                   No   
6                 4  ...          No  Steady                   No   
7                 5  ...          No      No                   No   
8                13  ...          No  Steady                   No   
9                12  ...          No  Steady                   No   

   Glipizide-metformin  Glimepiride-pioglitazone  Metformin-rosiglitazone  \
0                   No                        No                       No   
1                   No                        No                       No   
2                   No                        No                       No   
3                   No                        No                       No   
4                   No                        No                       No   
5                   No                        No                       No   
6                   No                        No                       No   
7                   No                        No                       No   
8                   No                        No                       No   
9                   No                        No                       No   

   Metformin-pioglitazone  Change Diabetes Medication Readmitted  
0                      No      No                  No         NO  
1                      No      Ch                 Yes        >30  
2                      No      No                 Yes         NO  
3                      No      Ch                 Yes         NO  
4                      No      Ch                 Yes         NO  
5                      No      No                 Yes        >30  
6                      No      Ch                 Yes         NO  
7                      No      No                 Yes        >30  
8                      No      Ch                 Yes         NO  
9                      No      Ch                 Yes         NO  

[10 rows x 50 columns]

We need to replace the "?" symbol with NaN so the dropna() can remove the missing values:


In [14]:
df1=df.replace('?',np.NaN)


We can drop missing values along the column "weight" as follows:


In [15]:
df=df1.dropna(subset=["Diagnosis2"], axis=0)
df.head(20)

Encounter Id  Patient Number             Race  Gender       Age Weight  \
1         149190        55629189        Caucasian  Female   [10-20)    NaN   
2          64410        86047875  AfricanAmerican  Female   [20-30)    NaN   
3         500364        82442376        Caucasian    Male   [30-40)    NaN   
4          16680        42519267        Caucasian    Male   [40-50)    NaN   
5          35754        82637451        Caucasian    Male   [50-60)    NaN   
6          55842        84259809        Caucasian    Male   [60-70)    NaN   
7          63768       114882984        Caucasian    Male   [70-80)    NaN   
8          12522        48330783        Caucasian  Female   [80-90)    NaN   
9          15738        63555939        Caucasian  Female  [90-100)    NaN   
10         28236        89869032  AfricanAmerican  Female   [40-50)    NaN   
11         36900        77391171  AfricanAmerican    Male   [60-70)    NaN   
12         40926        85504905        Caucasian  Female   [40-50)    NaN   
13         42570        77586282        Caucasian    Male   [80-90)    NaN   
14         62256        49726791  AfricanAmerican  Female   [60-70)    NaN   
15         73578        86328819  AfricanAmerican    Male   [60-70)    NaN   
16         77076        92519352  AfricanAmerican    Male   [50-60)    NaN   
17         84222       108662661        Caucasian  Female   [50-60)    NaN   
18         89682       107389323  AfricanAmerican    Male   [70-80)    NaN   
19        148530        69422211              NaN    Male   [70-80)    NaN   
20        150006        22864131              NaN  Female   [50-60)    NaN   

    Admission Type Id  Discharge Disposition Id  Admission Source Id  \
1                   1                         1                    7   
2                   1                         1                    7   
3                   1                         1                    7   
4                   1                         1                    7   
5                   2                         1                    2   
6                   3                         1                    2   
7                   1                         1                    7   
8                   2                         1                    4   
9                   3                         3                    4   
10                  1                         1                    7   
11                  2                         1                    4   
12                  1                         3                    7   
13                  1                         6                    7   
14                  3                         1                    2   
15                  1                         3                    7   
16                  1                         1                    7   
17                  1                         1                    7   
18                  1                         1                    7   
19                  3                         6                    2   
20                  2                         1                    4   

    Time In Hospital  ... Citoglipton Insulin  Glyburide-metformin  \
1                  3  ...          No      Up                   No   
2                  2  ...          No      No                   No   
3                  2  ...          No      Up                   No   
4                  1  ...          No  Steady                   No   
5                  3  ...          No  Steady                   No   
6                  4  ...          No  Steady                   No   
7                  5  ...          No      No                   No   
8                 13  ...          No  Steady                   No   
9                 12  ...          No  Steady                   No   
10                 9  ...          No  Steady                   No   
11                 7  ...          No  Steady                   No   
12          

Now, we have successfully read the raw dataset and added the correct headers into the dataframe.


 <div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #2: </h1>
<b>Find the name of the columns of the dataframe.</b>
</div>


In [16]:
# Write your code below and press Shift+Enter to execute

<details><summary>Click here for the solution</summary>

```python
    print(df.columns)
```
</details>


<h2>Save Dataset</h2>
<p>
Correspondingly, Pandas enables us to save the dataset to csv. By using the <code>dataframe.to_csv()</code> method, you can add the file path and name along with quotation marks in the brackets.
</p>
<p>
For example, if you would save the dataframe <b>df</b> as <b>diabetes.csv</b> to your local machine, you may use the syntax below, where <code>index = False</code> means the row names will not be written.
</p>


In [17]:
df.to_csv("diabetes.csv", index=False)

We can also read and save other file formats. We can use similar functions like **`pd.read_csv()`** and **`df.to_csv()`** for other data formats. The functions are listed in the following table:


<h2>Read/Save Other Data Formats</h2>

| Data Formate |        Read       |            Save |
| ------------ | :---------------: | --------------: |
| csv          |  `pd.read_csv()`  |   `df.to_csv()` |
| json         |  `pd.read_json()` |  `df.to_json()` |
| excel        | `pd.read_excel()` | `df.to_excel()` |
| hdf          |  `pd.read_hdf()`  |   `df.to_hdf()` |
| sql          |  `pd.read_sql()`  |   `df.to_sql()` |
| ...          |        ...        |             ... |


<h1 id="basic_insight">Basic Insight of Dataset</h1>
<p>
After reading data into Pandas dataframe, it is time for us to explore the dataset.<br>

There are several ways to obtain essential insights of the data to help us better understand our dataset.

</p>


<h2>Data Types</h2>
<p>
Data has a variety of types.<br>

The main types stored in Pandas dataframes are <b>object</b>, <b>float</b>, <b>int</b>, <b>bool</b> and <b>datetime64</b>. In order to better learn about each attribute, it is always good for us to know the data type of each column. In Pandas:

</p>


In [18]:
df.dtypes


Encounter Id                 int64
Patient Number               int64
Race                        object
Gender                      object
Age                         object
Weight                      object
Admission Type Id            int64
Discharge Disposition Id     int64
Admission Source Id          int64
Time In Hospital             int64
Payer Code                  object
Medical Specialty           object
Num Lab Procedures           int64
Num Procedures               int64
Num Medications              int64
Number Outpatient            int64
Number Emergency             int64
Number Inpatient             int64
Diagnosis1                  object
Diagnosis2                  object
Diagnosis3                  object
Number Diagnoses             int64
Max Glu Serum               object
A1c Result                  object
Metformin                   object
Repaglinide                 object
Nateglinide                 object
Chlorpropamide              object
Glimepiride         

A series with the data type of each column is returned.


In [19]:
# check the data type of data frame "df" by .dtypes
print(df.dtypes)

Encounter Id                 int64
Patient Number               int64
Race                        object
Gender                      object
Age                         object
Weight                      object
Admission Type Id            int64
Discharge Disposition Id     int64
Admission Source Id          int64
Time In Hospital             int64
Payer Code                  object
Medical Specialty           object
Num Lab Procedures           int64
Num Procedures               int64
Num Medications              int64
Number Outpatient            int64
Number Emergency             int64
Number Inpatient             int64
Diagnosis1                  object
Diagnosis2                  object
Diagnosis3                  object
Number Diagnoses             int64
Max Glu Serum               object
A1c Result                  object
Metformin                   object
Repaglinide                 object
Nateglinide                 object
Chlorpropamide              object
Glimepiride         

<p>
As shown above, it is clear to see that the data type of "EncounterId" and "PatientNbr" are <code>int64</code>, "Race" is <code>object</code>, etc.
</p>
<p>
These data types can be changed; we will learn how to accomplish this in a later module.
</p>


<h2>Describe</h2>
If we would like to get a statistical summary of each column e.g. count, column mean value, column standard deviation, etc., we use the describe method:


This method will provide various summary statistics, excluding <code>NaN</code> (Not a Number) values.


In [20]:
df.describe()

Encounter Id  Patient Number  Admission Type Id  \
count  1.014080e+05    1.014080e+05      101408.000000   
mean   1.653798e+08    5.441447e+07           2.025146   
std    1.026549e+08    3.869626e+07           1.446063   
min    1.252200e+04    1.350000e+02           1.000000   
25%    8.508277e+07    2.342905e+07           1.000000   
50%    1.525413e+08    4.562748e+07           1.000000   
75%    2.304963e+08    8.762756e+07           3.000000   
max    4.438672e+08    1.895026e+08           8.000000   

       Discharge Disposition Id  Admission Source Id  Time In Hospital  \
count             101408.000000        101408.000000     101408.000000   
mean                   3.719618             5.756163          4.401369   
std                    5.281755             4.066608          2.986794   
min                    1.000000             1.000000          1.000000   
25%                    1.000000             1.000000          2.000000   
50%                    1.000000             7.000000          4.000000   
75%                    4.000000             7.000000          6.000000   
max                   28.000000            25.000000         14.000000   

       Num Lab Procedures  Num Procedures  Num Medications  Number Outpatient  \
count       101408.000000   101408.000000    101408.000000      101408.000000   
mean            43.102201        1.342862        16.048892           0.370326   
std             19.684696        1.706917         8.122657           1.269159   
min              1.000000        0.000000         1.000000           0.000000   
25%             31.000000        0.000000        10.000000           0.000000   
50%             44.000000        1.000000        15.000000           0.000000   
75%             57.000000        2.000000        20.000000           0.000000   
max            132.000000        6.000000        81.000000          42.000000   

       Number Emergency  Number Inpatient  Number Diagnoses  
count     101408.000000     101408.000000     101408.000000  
mean           0.198199          0.636439          7.439117  
std            0.931316          1.262910          1.911536  
min            0.000000          0.000000          2.000000  
25%            0.000000          0.000000          6.000000  
50%            0.000000          0.000000          8.000000  
75%            0.000000          1.000000          9.000000  
max           76.000000         21.000000         16.000000

<p>
This shows the statistical summary of all numeric-typed (int, float) columns.<br>

For example, the attribute "DischargeDispositionId" has 3197 counts, the mean value of this column is 2.71, the standard deviation is 2.85, the minimum value is 1, 25th percentile is 1, 50th percentile is 1, 75th percentile is 3, and the maximum value is 18. <br>

However, what if we would also like to check all the columns including those that are of type object? <br><br>

You can add an argument <code>include = "all"</code> inside the bracket. Let's try it again.

</p>


In [21]:
# describe all the columns in "df" 
df.describe(include = "all")

Encounter Id  Patient Number       Race  Gender      Age    Weight  \
count   1.014080e+05    1.014080e+05      99157  101408   101408      3185   
unique           NaN             NaN          5       3       10         9   
top              NaN             NaN  Caucasian  Female  [70-80)  [75-100)   
freq             NaN             NaN      75839   54528    26033      1336   
mean    1.653798e+08    5.441447e+07        NaN     NaN      NaN       NaN   
std     1.026549e+08    3.869626e+07        NaN     NaN      NaN       NaN   
min     1.252200e+04    1.350000e+02        NaN     NaN      NaN       NaN   
25%     8.508277e+07    2.342905e+07        NaN     NaN      NaN       NaN   
50%     1.525413e+08    4.562748e+07        NaN     NaN      NaN       NaN   
75%     2.304963e+08    8.762756e+07        NaN     NaN      NaN       NaN   
max     4.438672e+08    1.895026e+08        NaN     NaN      NaN       NaN   

        Admission Type Id  Discharge Disposition Id  Admission Source Id  \
count       101408.000000             101408.000000        101408.000000   
unique                NaN                       NaN                  NaN   
top                   NaN                       NaN                  NaN   
freq                  NaN                       NaN                  NaN   
mean             2.025146                  3.719618             5.756163   
std              1.446063                  5.281755             4.066608   
min              1.000000                  1.000000             1.000000   
25%              1.000000                  1.000000             1.000000   
50%              1.000000                  1.000000             7.000000   
75%              3.000000                  4.000000             7.000000   
max              8.000000                 28.000000            25.000000   

        Time In Hospital  ... Citoglipton Insulin  Glyburide-metformin  \
count      101408.000000  ...      101408  101408               101408   
unique               NaN  ...           1       4                    4   
top                  NaN  ...          No      No                   No   
freq                 NaN  ...      101408   47282               100705   
mean            4.401369  ...         NaN     NaN                  NaN   
std             2.986794  ...         NaN     NaN                  NaN   
min             1.000000  ...         NaN     NaN                  NaN   
25%             2.000000  ...         NaN     NaN                  NaN   
50%             4.000000  ...         NaN     NaN                  NaN   
75%             6.000000  ...         NaN     NaN                  NaN   
max            14.000000  ...         NaN     NaN                  NaN   

        Glipizide-metformin  Glimepiride-pioglitazone  \
count                101408                    101408   
unique                    2                         2   
top                      No                        No   
freq                 101395                    101407   
mean                    NaN                       NaN   
std                     NaN                       NaN   
min                     NaN                       NaN   
25%                     NaN                       NaN   
50%                     NaN                       NaN   
75%                     NaN                       NaN   
max                     NaN                       NaN   

        Metformin-rosiglitazone  Metformin-pioglitazone  Change  \
count                    101408                  101408  101408   
unique                        2                       2       2   
top                          No                      No      No   
freq                     101406                  101407   54553   
mean                        NaN                     NaN     NaN   
std                         NaN                     NaN     NaN   
min                         NaN                     NaN     NaN   
25%                         NaN                 

<p>
Now it provides the statistical summary of all the columns, including object-typed attributes.<br>

We can now see how many unique values there, which one is the top value and the frequency of top value in the object-typed columns.<br>

Some values in the table above show as "NaN". This is because those numbers are not available regarding a particular column type.<br>

</p>

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #3: </h1>

<p>
You can select the columns of a dataframe by indicating the name of each column. For example, you can select the three columns as follows:
</p>
<p>
    <code>dataframe[[' column 1 ',column 2', 'column 3']]</code>
</p>
<p>
Where "column" is the name of the column, you can apply the method  ".describe()" to get the statistics of those columns as follows:
</p>
<p>
    <code>dataframe[[' column 1 ',column 2', 'column 3'] ].describe()</code>
</p>

Apply the  method to ".describe()" to the columns 'Age' and 'Weight'.

</div>


In [22]:
# Write your code below and press Shift+Enter to execute

<details><summary>Click here for the solution</summary>

```python
    df[['Age', 'Weight']].describe()
```
</details>

<h2>Info</h2>
Another method you can use to check your dataset is:


It provides a concise summary of your DataFrame.

This method prints information about a DataFrame including the index dtype and columns, non-null values and memory usage.


In [23]:
# look at the info of "df"
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 101408 entries, 1 to 101765
Data columns (total 50 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   Encounter Id              101408 non-null  int64 
 1   Patient Number            101408 non-null  int64 
 2   Race                      99157 non-null   object
 3   Gender                    101408 non-null  object
 4   Age                       101408 non-null  object
 5   Weight                    3185 non-null    object
 6   Admission Type Id         101408 non-null  int64 
 7   Discharge Disposition Id  101408 non-null  int64 
 8   Admission Source Id       101408 non-null  int64 
 9   Time In Hospital          101408 non-null  int64 
 10  Payer Code                61314 non-null   object
 11  Medical Specialty         51557 non-null   object
 12  Num Lab Procedures        101408 non-null  int64 
 13  Num Procedures            101408 non-null  int64 
 14  Num 

<h1>Excellent! You have just completed the  Introduction Notebook!</h1>


### Thank you for completing this lab!

## Author

<a href="https://www.linkedin.com/in/joseph-s-50398b136/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01" target="_blank">Joseph Santarcangelo</a>

### Other Contributors

<a href="https://www.linkedin.com/in/mahdi-noorian-58219234/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01" target="_blank">Mahdi Noorian PhD</a>

Bahare Talayian

Eric Xiao

Steven Dong

Parizad

Hima Vasudevan

<a href="https://www.linkedin.com/in/fiorellawever/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDA0101ENSkillsNetwork20235326-2021-01-01" target="_blank">Fiorella Wenver</a>

<a href="https:// https://www.linkedin.com/in/yi-leng-yao-84451275/ " target="_blank" >Yi Yao</a>.

## Change Log

| Date (YYYY-MM-DD) | Version | Changed By | Change Description                                         |
| ----------------- | ------- | ---------- | ---------------------------------------------------------- |
| 2022-08-23        | 2.4     | Malika     | Import micropip added and parameter for ipwidgets and tqdm |
| 2020-10-30        | 2.3     | Lakshmi    | Changed URL of the csv                                     |
| 2020-09-22        | 2.2     | Nayef      | Added replace() method to remove '?'                       |
| 2020-09-09        | 2.1     | Lakshmi    | Made changes in info method of dataframe                   |
| 2020-08-27        | 2.0     | Lavanya    | Moved lab to course repo in GitLab                         |

<hr>

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
